internal/proto/encoding.go

```diff
@@ -333,3 +333,27 @@ func MarshalToRepeatedBytes[T Bytes](b []byte, num protowire.Number, v []T) int
 	}
 	return off
 }
+
+// SizeRepeatedMessages returns the encoded size of 'repeated M' protobuf field
+// with given number and values.
+func SizeRepeatedMessages[M Message](num protowire.Number, v []M) int {
```

Comment:
cuz it accepts slice

Reply:
i meant you did a generic func with `[]M` type of `Message` interface to pass it to func that accept `M` _interface_. i do not understand why generics are needed here

Reply:
I think it's about slices of particular types vs slices of interfaces. Conversion from specific type to interface happens here, without generics it'd be the caller who'd need to do this conversion.

Reply:
yep caller can skip `[]T -> []M`, this is one of most powerful things came with generics. If u asking why `M` is not a generic type in `SizeEmbedded`? Well, it can be, mb it's even better to, but this will be a fully compatible change so doesnt rly matter for now

Reply:
huh, we never get slices of `Message` but try to accept them via a generic func, got it. the commit is fully about refactoring the code but it made it harder to me personally. as for me, i always try to make a stupid helper func in such cases to convert args separately from main func like `toMessageSlice[T Message](v []M) []Message`